GeneratorInterface/GenFilters/src/MCSingleParticleFilter.cc

```diff
@@ -95,8 +99,9 @@ bool MCSingleParticleFilter::filter(edm::Event& iEvent, const edm::EventSetup& i
      for (unsigned int i = 0; i < particleID.size(); i++){
        if (particleID[i] == (*p)->pdg_id() || particleID[i] == 0) {
     
-	 if ( (*p)->momentum().perp() > ptMin[i] && (*p)->momentum().eta() > etaMin[i] 
-	      && (*p)->momentum().eta() < etaMax[i] && ((*p)->status() == status[i] || status[i] == 0)) { 
+         HepMC::FourVector mom = zboost((*p)->momentum());
+	 if ( mom.perp() > ptMin[i] && mom.eta() > etaMin[i] 
```

Comment:
call the boost function only for particles that pass the pt and status requirements.